src/main/java/pl/piekoszek/gorskimatches/equation/SolutionToQuizzesMapper.java

```diff
@@ -52,4 +104,33 @@ public Map<String, Set<String>> insideSingleNumber(String solution) {
         }
         return quizzesAndSolutions;
     }
+
+    public Map<String, Set<String>> insideEquation(String solution) {
+        Map<String, Set<String>> quizzesAndSolutionsWithinEquation = new HashMap<>();
+
+        for (int i = 0; i < solution.length(); i++) {
+            char numberOrSymbolToBeReplaced = solution.charAt(i);
+            var numbersAndSymbolsToBeTakenFrom = takeOneMatchFromNumberOrSymbol.get(numberOrSymbolToBeReplaced);
+
+            for (Character numberWithTakenMatch : numbersAndSymbolsToBeTakenFrom) {
+                String quizWithTakenMatch = characterChanger.changeCharactersInString(solution, i, numberWithTakenMatch);
+                for (int k = 0; k < quizWithTakenMatch.length(); k++) {
+                    if (k != i) {
+                        char numberOrSymbolToAddMatchTo = quizWithTakenMatch.charAt(k);
+                        var numbersAndSymbolsToBeAddedTo = giveOneMatchFromNumberOrSymbol.get(numberOrSymbolToAddMatchTo);
```

Comment:
I'm not sure if it's `numbersAndSymbolsToBeAddedTo`. I think it's something after adding.